sdks/java/core/src/test/java/org/apache/beam/sdk/io/AvroIOTest.java

```diff
@@ -212,6 +276,45 @@ public void testAvroIOWriteAndReadSchemaUpgrade() throws Throwable {
     p.run();
   }
 
+  @Test
+  public void testWriteWithDefaultCodec() throws Exception {
+    AvroIO.Write.Bound<GenericRecord> write = AvroIO.Write
+        .to("gs://bucket/foo/baz");
+    assertTrue(write.getCodec().toString().equals(CodecFactory.deflateCodec(6).toString()));
+  }
+
+  @Test
+  public void testWriteWithCustomCodec() throws Exception {
+    AvroIO.Write.Bound<GenericRecord> write = AvroIO.Write
+        .to("gs://bucket/foo/baz")
+        .withCodec(CodecFactory.snappyCodec());
+    assertTrue(write.getCodec().toString().equals(SNAPPY_CODEC));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testWriteWithSerDeCustomDeflateCodec() throws Exception {
+    AvroIO.Write.Bound<GenericRecord> write = AvroIO.Write
+        .to("gs://bucket/foo/baz")
+        .withCodec(CodecFactory.deflateCodec(9));
+
+    AvroIO.Write.Bound<GenericRecord> serdeWrite = SerializableUtils.clone(write);
+
+    assertTrue(serdeWrite.getCodec().toString().equals(CodecFactory.deflateCodec(9).toString()));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testWriteWithSerDeCustomXZCodec() throws Exception {
+    AvroIO.Write.Bound<GenericRecord> write = AvroIO.Write
+        .to("gs://bucket/foo/baz")
+        .withCodec(CodecFactory.xzCodec(9));
+
+    AvroIO.Write.Bound<GenericRecord> serdeWrite = SerializableUtils.clone(write);
+
+    assertTrue(serdeWrite.getCodec().toString().equals(CodecFactory.xzCodec(9).toString()));
+  }
```

Comment:
*This is **completely optional**. I'm only including this nitpick because there were other actual comments*

We'll get slightly better error message if the code was written as:

```
assertEquals(CodecFactory.xzCodec(9), serdWrite.getCodec())`
```

(with `toString()` as it is now, if those equality checks don't work).

There are a few other optional instances of opportunities for improvement in this file and the other test file.